articles/140_topic_and_service_name_mapping.md

```diff
@@ -243,41 +221,72 @@ Here is a non-exhaustive list of prefixes:
 | ROS Parameter        | rp     |
 | ROS Action           | ra     |
 
-While all planned prefixes consist of two characters, i.e. `rX`, anything proceeding the first namespace separator, i.e. `__`, can be considered part of the prefix.
+While all planned prefixes consist of two characters, i.e. `rX`, anything proceeding the first namespace separator, i.e. `/`, can be considered part of the prefix.
 The standard reserves the right to use up to 8 characters for the prefix in case additional prefix space is needed in the future.
 
-### Communicating with Non-ROS Topics
+#### Communicating with Non-ROS Topics
 
 Since all ROS topics are prefixed when being converted to DDS topic names, it makes it impossible to subscribe to existing DDS topics which do not follow the same naming pattern.
 For example, if an existing DDS program is publishing on the `image` topic (and is using the DDS equivalent to the ROS message type) then a ROS program could not subscribe to it because of the name mangling.
 Therefore to allow ROS programs to interoperate with "native" DDS topic names the API should provide a way to skip the ROS specific prefixing.
 
-### ROS to DDS Name Conversion Examples
+### DDS Partitions as namespace hierarchy
 
-Here are some examples of translations between ROS topic names and the corresponding DDS topic names:
+We chose to implement the ROS namespace hierarchy with the help of DDS Partitions.
+DDS Paritions are part of the Subscription/Publisher QoS settings and have been considered as the prefered method of implementing namespaces.
+A subscriber and publisher not only match when their basename topic matches, they also have to be specified on the same partition.
+The big benefit of handling the namespaces with partitions is that it is an official QoS within the DDS specification.
+Furthermore, we can separate a ROS topic hierarchy from a DDS implementation, by forwarding the ROS topic in form of `/ns1/ns2/ns3/base_name` to the DDS implementation, which then takes care about setting up a hierarchy.
+We can therefore make the translation from a ROS Topic (namespaces and basename separated by forward slashes) to a DDS Topic (topic plus paritions) an implementation detail of each RMW implementation.
+This further upholds the flexibility of each alternative to DDS (e.g. ZeroMQ) to realize a different mapping depending on what the middleware provides.
```

Comment:
This paragraph is a bit confusing to me, it might just be the wording. Would you mind if I open a pr against this with a different wording?